internal/common/optimisation/optimisation.go

```diff
@@ -0,0 +1,8 @@
+package optimisation
+
+import "gonum.org/v1/gonum/mat"
+
+type Optimiser interface {
+	Update(out, p *mat.VecDense, g mat.Vector) *mat.VecDense
```

Comment:
Better names for parameters? It's not clear what these do from the signature.

Reply:
Fixed. I left p and g as-is in the implementations as I think the long names make the code a bit less readable there.